src/GovUk.Education.ExploreEducationStatistics.Data.Services/MetaGuidanceSubjectService.cs

```diff
@@ -34,16 +34,24 @@ public class MetaGuidanceSubjectService : IMetaGuidanceSubjectService
             _persistenceHelper = persistenceHelper;
         }
 
-        public async Task<Either<ActionResult, List<MetaGuidanceSubjectViewModel>>> GetSubjects(Guid releaseId)
+        public async Task<Either<ActionResult, List<MetaGuidanceSubjectViewModel>>> GetSubjects(Guid releaseId,
+            List<Guid> subjectIds = null)
         {
             return await _persistenceHelper.CheckEntityExists<Release>(releaseId)
                 .OnSuccess(async release =>
                 {
-                    var releaseSubjects = await _context
+                    var releaseSubjectsQueryable = _context
                         .ReleaseSubject
                         .Include(s => s.Subject)
-                        .Where(s => s.ReleaseId == releaseId)
-                        .ToListAsync();
+                        .Where(rs => rs.ReleaseId == releaseId);
+
+                    if (subjectIds != null)
```

Comment:
Dunno how EF handles empty collections when converting the LINQ expression, but probably be better to check that `subjectIds` is not empty as well to ensure there are no SQL `WHERE IN ()` errors.

Reply:
I'm not sure that when this is called with an empty list that it should just default to returning all the Subjects.

I did it like this in case the data files query in `MetaGuidanceService.BuildViewModel` finds no files and calls it with an empty subject list. If for some reason we have ReleaseSubjects in the statistics database they would be returned because they match the Release, but there would be no files.  

I know this should never happen right now, but ReleaseSubjects can exist that won't be in the data files list, for example replacement subjects.

EF does handle the empty list in this situation.

I considered the following options when implementing it (going with the assumption that the empty list and null list should be treated differently).

-  a new method where we can give the two methods clear names like `GetSubjects(Guid releaseId)` and `GetAllSubjects(Guid releaseId, List<Guid> subjectIds)`.
- an overload with a `List<Guid> subjectIds` param.
- using `params Guid[] subjectIds`.
- `List<Guid> subjectIds = null`.

I ruled out `params` because we can't differentiate between null and empty.

The first two are clearer to the caller but unless we duplicate the LINQ query in each one it would be implemented in the same way as this PR but with an new `GetAllSubjects` method that will call `GetSubjects` with a null list.

Reply:
Ah okay, I was mostly just concerned about EF being able to handle empty lists correctly. In other ORMs it's pretty commonplace for empty lists in `WHERE IN`s to throw SQL exceptions.

Think it's fine to leave it as is :wink:

Reply:
Okay thanks for reviewing it. 👍